docs/content/python/guides/rbac.md

```diff
@@ -0,0 +1,326 @@
+For example, if we grant the `"push"` permission to the `"maintainer"` role in
+the `Repository` resource block, then a user who's been assigned the
+`"maintainer"` role for a particular repository can `"push"` to that
+repository. Here's our `Repository` resource block with a few shorthand rules
+added:
+
+```polar
+resource Repository {
+  permissions = [ "read", "push" ];
+  roles = [ "contributor", "maintainer" ];
+
+  "push" if "maintainer";
+  "read" if "maintainer";
+  "read" if "contributor";
+}
+```
+
+Shorthand rules expand to regular [Polar rules](polar-syntax#rules) when a
+policy is loaded. The `"push" if "maintainer"` shorthand rule above expands
+to:
+
+```polar
+has_permission(actor: Actor, "push", repository: Repository) if
+  has_role(actor, "maintainer", repository);
+```
+
+{{% callout "Note" "blue" %}}
+  Instances of our application's `User` class will match the `Actor`
+  [specializer](polar-syntax#specialization) because of our `actor User {}`
+  resource block declaration.
+{{% /callout %}}
+
+<!-- TODO(gj): transition -->
+
+## Grant roles to other roles
+
+All of the shorthand rules we've written so far have been in the `<permission>
+if <role>` form, but we can also write `<role1> if <role2>` rules. This type
+of rule is great for situations where you want to express that `<role2>` should
+be granted every permission you've granted to `<role1>`.
+
+In the previous snippet, the permissions granted to the `"maintainer"` role are
+a superset of those granted to the `"contributor"` role. If we replace the
+existing `"read" if "maintainer"` rule with `"contributor" if "maintainer"`,
+the `"maintainer"` role still grants the `"read"` permission:
+
+```polar
+resource Repository {
+  permissions = [ "read", "push" ];
+  roles = [ "contributor", "maintainer" ];
+
+  "push" if "maintainer";
+  "read" if "contributor";
+
+  "contributor" if "maintainer";
+}
+```
+
+In addition, any permissions we grant the `"contributor"` role in the future
+will automatically propagate to the `"maintainer"` role.
+
+<!-- TODO(gj): better heading -->
+## Access role assignments stored in the application
+
+An Oso policy contains authorization *logic*, but the application remains in
+control of all authorization *data*. For example, the logic that the
+`"maintainer"` role on a repository grants the `"push"` permission lives in the
+policy, but Oso doesn't manage the data of which users have been assigned the
+`"maintainer"` role for `Repository(id=2)`. That data stays in the application,
+and Oso asks the application for it as needed.
+
+The main question Oso asks is: does `User(id=1)` have the `"maintainer"` role
+on `Repository(id=2)`? For Oso to be able to ask this question, we need to
+implement a `has_role()` rule in the policy:
+
+```polar
+has_role(user: User, name: String, resource: Resource) if
+  role in user.roles and
+  role matches { name: name, resource: resource };
```

Comment:
Not sure about this vs two separate lines because the `matches` operator is less obvious than `=`.

Reply:
1e5c9deb